SIGSEGV on azure role unassignments

We where in the middle of upgrading our vault cluster from 1.15.6 to 1.16.1 when the servers started failing. It seems like there is an issue when azure roles are being unassigned and it causes the vault service to crash.


I suspect the root of this issue is this change from the REST API to the graph API client and that `rawResponse` is still `nil` when we are peeking at the `statusCode`.


> Replace go-autorest MS Graph client with msgraph-sdk-go (https://github.com/hashicorp/vault-plugin-secrets-azure/pull/169)
https://github.com/hashicorp/vault-plugin-secrets-azure/commit/416c8fd8f567360cc8187bbc20c0d3fdb5c68d7e#diff-4b667feae66c9d46b21b9ecc19e8958cf4472d162ce0a47ac3e8386af8bbd8cfL213-R221


Traceback:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x6b18f07]
goroutine 52825 [running]:
github.com/hashicorp/vault-plugin-secrets-azure.(*client).unassignRoles(0xc00ab0eaa0, {0xc9c2b68, 0xc0089a1960}, {0xc00fb0c800?, 0x2e, 0xc00f888910?})
        /home/runner/go/pkg/mod/github.com/hashicorp/vault-plugin-secrets-azure@v0.17.0/client.go:221 +0x127
github.com/hashicorp/vault-plugin-secrets-azure.(*azureSecretBackend).rollbackRoleAssignWAL(0xc0038906c0, {0xc9c2b68, 0xc0089a1960}, 0xc0054030e0, {0x8cc67c0, 0xc00fa3be30})
        /home/runner/go/pkg/mod/github.com/hashicorp/vault-plugin-secrets-azure@v0.17.0/wal.go:157 +0x4ff
github.com/hashicorp/vault-plugin-secrets-azure.(*azureSecretBackend).walRollback(0x10f45f4?, {0xc9c2b68?, 0xc0089a1960?}, 0x3504?, {0xc00f6faad0?, 0xc00f9f9600?}, {0x8cc67c0

        /home/runner/go/pkg/mod/github.com/hashicorp/vault-plugin-secrets-azure@v0.17.0/wal.go:33 +0x94
github.com/hashicorp/vault/sdk/framework.(*Backend).handleWALRollback(0xc0062b0e10, {0xc9c2b68, 0xc0089a1960}, 0xc0054030e0)
        /home/runner/work/vault/vault/sdk/framework/backend.go:714 +0x3c3
github.com/hashicorp/vault/sdk/framework.(*Backend).handleRollback(0xc0062b0e10, {0xc9c2b68, 0xc0089a1960}, 0x270f?)
        /home/runner/work/vault/vault/sdk/framework/backend.go:657 +0xc6
github.com/hashicorp/vault/sdk/framework.(*Backend).HandleRequest(0xc0062b0e10, {0xc9c2b68, 0xc0089a1960}, 0xc0054030e0)
        /home/runner/work/vault/vault/sdk/framework/backend.go:219 +0x105
github.com/hashicorp/vault/builtin/plugin/v5.(*backend).HandleRequest(0xc0049a4280, {0xc9c2b68, 0xc0089a1960}, 0xc0054030e0)
        /home/runner/work/vault/vault/builtin/plugin/v5/backend.go:94 +0xbd
github.com/hashicorp/vault/vault.(*Router).routeCommon(0xc002ad2660, {0xc9c2b68, 0xc0089a1960}, 0xc0054030e0, 0x0)
        /home/runner/work/vault/vault/vault/router.go:803 +0x1686
github.com/hashicorp/vault/vault.(*Router).Route(...)
        /home/runner/work/vault/vault/vault/router.go:572
github.com/hashicorp/vault/vault.(*RollbackManager).attemptRollback(0xc003cbc840, {0xc9c2ac0, 0xc0039cb770}, {0xc00621ce66, 0x6}, 0xc003891080, 0x1)
        /home/runner/work/vault/vault/vault/rollback.go:341 +0x84d
github.com/hashicorp/vault/vault.(*RollbackManager).startOrLookupRollback.func1()
        /home/runner/work/vault/vault/vault/rollback.go:248 +0x35
github.com/gammazero/workerpool.worker(0x0?, 0x0?, 0x0?)
        /home/runner/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:237 +0x22
created by github.com/gammazero/workerpool.(*WorkerPool).dispatch in goroutine 3248
        /home/runner/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:197 +0x2b3
vault.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
vault.service: Failed with result 'exit-code'.
vault.service: Consumed 20.741s CPU time.
vault.service: Scheduled restart job, restart counter is at 6.
```

--- a/client.go
+++ b/client.go
@@ -218,7 +218,7 @@ func (c *client) unassignRoles(ctx context.Context, roleIDs []string) error {
 		ctxWithResp := policy.WithCaptureResponse(ctx, &rawResponse)
 		if _, err := c.provider.DeleteRoleAssignmentByID(ctxWithResp, id); err != nil {
 			// If a role was deleted manually then Azure returns a error and status 204
-			if rawResponse.StatusCode == http.StatusNoContent || rawResponse.StatusCode == http.StatusNotFound {
+			if rawResponse != nil && rawResponse.StatusCode == http.StatusNoContent || rawResponse.StatusCode == http.StatusNotFound {
 				continue
 			}
 

